Mozu.Api/ApiContext.cs

```diff
@@ -212,7 +217,8 @@ public IApiContext CloneWith(Action<IApiContext> contextModification)
                 NoCacheUpdate = (!String.IsNullOrEmpty(this.NoCacheUpdate) ? (string)this.NoCacheUpdate.Clone() : null),
                 BypassCache = (!String.IsNullOrEmpty(this.BypassCache) ? (string)this.BypassCache.Clone() : null),
                 Pricelist = (!String.IsNullOrEmpty(this.Pricelist) ? (string)this.Pricelist.Clone() : null),
-                PreviewDate = (!String.IsNullOrEmpty(this.PreviewDate) ? (string)this.PreviewDate.Clone() : null)
+                PreviewDate = (!String.IsNullOrEmpty(this.PreviewDate) ? (string)this.PreviewDate.Clone() : null),
+				CustomHeaders= (this.CustomHeaders != null) ? (new Dictionary<string,string>(this.CustomHeaders)) : null
```

Comment:
what is this doing here? not cloning the custom headers?

Reply:
add a test to validate the clone is working.

Reply:
Added tests to validate apiContext cloning.